File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const myMap =  new Map()

    for(let i=0;i<nums.length;i++){
        let value = target - nums[i]

        if(myMap.has(value)){
          return [myMap.get(value),i]
        }

        myMap.set(nums[i],i)
    }

    return []
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapping = {}

        for index, val in enumerate(nums):
            diff = target - val
            if diff in mapping:
                return [index, mapping[diff]]
            else:
                mapping[val] = index